### PR TITLE
STORM-1535: Make sure hdfs key tab login happens only once for multip…

### PR DESCRIPTION
**Pull request by priyank5485**
…le bolts/executors.

### CODE REVIEW

**Comment by harshach** [comment]
+1

File: external/storm-hdfs/src/main/java/org/apache/storm/hdfs/common/security/HdfsSecurityUtil.java [hunk 1]
```diff
@@ -40,23 +41,27 @@
...
-                LOG.info("Logging in using keytab as AutoHDFS is not specified for " + TOPOLOGY_AUTO_CREDENTIALS);
-                String keytab = (String) conf.get(STORM_KEYTAB_FILE_KEY);
-                if (keytab != null) {
-                    hdfsConfig.set(STORM_KEYTAB_FILE_KEY, keytab);
-                }
-                String userName = (String) conf.get(STORM_USER_NAME_KEY);
-                if (userName != null) {
-                    hdfsConfig.set(STORM_USER_NAME_KEY, userName);
+                // compareAndSet added because of https://issues.apache.org/jira/browse/STORM-1535
+                // need to test it first during ERIE release testing since the JIRA says "might" be and port it to apache.
```
Comment (satishd) [inline external/storm-hdfs/src/main/java/org/apache/storm/hdfs/common/security/HdfsSecurityUtil.java]: @priyank5485 You may want to remove this comment as it is not valid here in apache repo.

**Comment by priyank5485** [comment]
Thanks for catching that @satishd Removed the comment.

**Comment by ptgoetz** [comment]
+1

**Comment by ptgoetz** [comment]
Thanks @priyank5485. Merged to master, 1.x-branch and 0.10.x-branch.

**Pull request merged**